### PR TITLE
Add Auto-Update & Build

### PR DESCRIPTION
**Pull request by adamrusted**
Copying a load of code from the website repo - this should hopefully allow us to trigger an update and release through the GitHub API. Could a @simple-icons/maintainers with owner privileges set up the `RELEASE_TOKEN` secret ~and anything relevant on the main repo~ though?

### CODE REVIEW

File: .github/workflows/auto-release.yml
Comment (developStorm) [inline .github/workflows/auto-release.yml:37]: Would be somewhat better to use 
```bash
git config --global user.email "github-actions[bot]@users.noreply.github.com"
git config --global user.name "github-actions[bot]"
```
A commit created with this information will be linked to https://github.com/apps/github-actions instead of a specific user on GitHub.

Current information will be linked to this user: https://github.com/actions-user (not sure who actually owns it)

Comment (ericcornelissen) [reply .github/workflows/auto-release.yml:37]: I like the suggestions @developStorm :+1:

If I recall correctly I actually copied the current approach from some official GitHub Actions example, I'd guess they (or at least someone) create the account for this purpose, but I think it's better to change it to your suggestion. Can you update this @adamrusted ?

Also, as I mentioned I originally wrote _this_ workflow for other simple-icons projects. So, if you want @devlopStorm, feel free to open Pull Requests against all simple-icons projects workflow files that make a commit like this.

**Comment by ericcornelissen** [comment]
This depends on #4

File: package.json
Comment (ericcornelissen) [inline package.json:15]: I think this should be renamed to `prebuild`. I'd say `clean` is generally reserved for cleaning up generated artefacts that are not part of version control. Moreover, this command should only ever be run before `build` so naming it as such makes sense.

If this is updated you can remove the `npm run clean` command from the workflow file as well.

File: .github/workflows/auto-release.yml
Comment (ericcornelissen) [inline .github/workflows/auto-release.yml:39]: I would prefer this with the extra enter:

```suggestion
          git config --global user.name "github-actions[bot]"

          # Create a commit for the simple-icons dependency update
```

Alternatively, we could split this up into two steps, "Set up git credentials" and "Commit updates" but then we _should_ update it in every repository.

Either solution works for me.

File: .github/workflows/auto-release.yml
Comment (ericcornelissen) [inline .github/workflows/auto-release.yml:47]: I would prefer this with the extra enter:

```suggestion
          git remote add origin https://${{secrets.RELEASE_TOKEN}}@github.com/simple-icons/simple-icons-pdf.git

          # Push updates to origin
```

Alternatively, we could split this up into two steps, "Set remote" and "Push updates" but then we _should_ update it in every repository.

Either solution works for me.

**Review by ericcornelissen — CHANGES_REQUESTED** [review]

**Comment by ericcornelissen** [comment]
Thanks for the update @adamrusted, I'm experiencing some trouble running this locally as it appears the icons are not generated with the `prebuild` step. It seems to be because the `icons/` directory doesn't exist because if I add `&& mkdir icons` to the `prebuild` step it works fine again.

Do you experience the same? If so, that might be due to a difference in OS - I'm testing this on Ubuntu.

**Comment by adamrusted** [comment]
I'll be honest, I haven't tested it (I'm on Windows, without WSL installed).
I imagine it'd probably be easier to just `rm` the files within the folder, without deleting the folder itself?

**Review by mondeja — COMMENTED** [review]
Great work! 👍🏼 

I'm wondering if you're thinking about adding here the "release to NPM and Github" logic here, or go with that in another PR?

As a remainder: we must be careful here, creating the trigger to launch this pull [in main repository publish workflow](https://github.com/simple-icons/simple-icons/blob/f9e8ecb77756002c147da36a436a094817e9bc47/.github/workflows/publish.yml#L104-L114) before the first execution.

**Comment by ericcornelissen** [comment]
> I'm wondering if you're thinking about adding here the "release to NPM and Github" logic here, or go with that in another PR?

I think we should decide first whether we want to publish to npm and whether it we want to create GitHub releases. Unless we decide that before this is merged, I think we should just go ahead and merge this.


> As a remainder: we must be careful here, creating the trigger to launch this pull in main repository publish workflow before the first execution.

Definitely :+1:

**Review by mondeja — APPROVED** [review]
> I think we should decide first whether we want to publish to npm and whether it we want to create GitHub releases. Unless we decide that before this is merged, I think we should just go ahead and merge this.

My opinion is to follow the same strategy of [simple-icons-font](https://github.com/simple-icons/simple-icons-font). What do you think?

**Comment by ericcornelissen** [comment]
I'm fine either way. I'm not sure how useful it is to have the PDFs available as an npm package, but on the other hand it probably doesn't hurt either.

**Comment by mondeja** [comment]
> I'm not sure how useful it is to have the PDFs available as an npm package, but on the other hand it probably doesn't hurt either.

It's a rare usage case, but with Webpack [file-loader](https://www.npmjs.com/package/file-loader) you can import any type of file, [even PDFs](https://www.py4u.net/discuss/293830).

**Comment by ericcornelissen** [comment]
> It's a rare usage case, but with Webpack [file-loader](https://www.npmjs.com/package/file-loader) you can import any type of file, [even PDFs](https://www.py4u.net/discuss/293830).

Interesting, that's a pretty good argument to just go ahead and release this to npm.

@adamrusted, if you want you can include npm (and GitHub) releases here as well. If not, we can take care of that in a separate PR as well.

**Comment by adamrusted** [comment]
I have no experience setting up npm and Github releases, and quite frankly no time to look into it at this time. @ericcornelissen or @mondeja - if you're able to continue the work in this PR to get it up and running then by all means go for it. I agree that the use-case for a PDF npm module might be very limited, but it would also help us tidy up the modules a fair bit.

**Comment by jorgeamadosoria** [comment]
Going by the comments, this is ready for merging.
For what little I understand of the yml code, this seems complete.
What's left to get this merged so it can be triggered from the main repo, @ericcornelissen @adamrusted ?

From what I see, if this is merged and a trigger is added from the main repo (similar to the one for simple-icons-font) then the problem of the pdfs not being in synced is solved, and the svg-to-pdf-bot project can be deleted. 
Also, there are a handful of tickets that can be closed in a couple repositories as well.
So.... what's missing?

**Comment by ericcornelissen** [comment]
> So.... what's missing?

The remaining problem is that the website depends on `simple-icons` and `simple-icons-pdf`, but is triggered by a rebuild from `simple-icons` after which we can only be sure that `simple-icons` has been updated. There have been two proposals to solve this (neither of which I'm a particular fan of personally, but that doesn't mean we can't implement them):

1. Implement a sleep/wait/timeout before `curl`ing the website's release workflow from `simple-icons` (- not a fan because it may fail sporadically without notice)
2. Move the `curl` request to the website's release workflow to this project (- not a fan because I'd like to keep the releases coordinated in a single file so it's easier to understand).

I haven't found the time yet to brainstorm a "proper" solution. Any suggestion is welcome, including proposing to just go with one of the above two for the time being.


> the svg-to-pdf-bot project can be deleted

This project is already archived, but if you want to delete it that also works.

**Comment by jorgeamadosoria** [comment]
> > So.... what's missing?
> 
> The remaining problem is that the website depends on `simple-icons` and `simple-icons-pdf`, but is triggered by a rebuild from `simple-icons` after which we can only be sure that `simple-icons` has been updated. There have been two proposals to solve this (neither of which I'm a particular fan of personally, but that doesn't mean we can't implement them):
> 
> 1. Implement a sleep/wait/timeout before `curl`ing the website's release workflow from `simple-icons` (- not a fan because it may fail sporadically without notice)
> 2. Move the `curl` request to the website's release workflow to this project (- not a fan because I'd like to keep the releases coordinated in a single file so it's easier to understand).
> 
> I haven't found the time yet to brainstorm a "proper" solution. Any suggestion is welcome, including proposing to just go with one of the above two for the time being.
> 
> > the svg-to-pdf-bot project can be deleted
> 
> This project is already archived, but if you want to delete it that also works.

What about periodically triggering the pdf repo, say once a day? Check if there have being commits to master in the last day, and if they are, rebuild the pdfs, for example. Sure, that won't guarantee that 100% of the icons will have the pdf download ready, but the missing pdfs will be at most the ones committed during the day, say 4 or 5 out of 2k. I think that makes it negligible.

Or better yet, what about, again, dynamically geerating the pdfs from the website? yes, the website would own the code that creates the PDFs, coupling, etc., but I saw the code that generate the PDFs. It's extremely basic, since the icon is just translated to a PDF document without any frills. And it would remove the need for the repository, the npm package discussed here, the workflow... huge simplification of the codebase. I still think this is the best choice, by far.

**Comment by ericcornelissen** [comment]
> What about periodically triggering the pdf repo, say once a day? Check if there have being commits to master in the last day, and if they are, rebuild the pdfs, for example. Sure, that won't guarantee that 100% of the icons will have the pdf download ready, but the missing pdfs will be at most the ones committed during the day, say 4 or 5 out of 2k. I think that makes it negligible.

This is pointless as this project generates pdfs based on the SVGs available on NPM, not what has been contributed on GitHub.


> Or better yet, what about, again, dynamically geerating the pdfs from the website? yes, the website would own the code that creates the PDFs, coupling, etc., but I saw the code that generate the PDFs. It's extremely basic, since the icon is just translated to a PDF document without any frills. And it would remove the need for the repository, the npm package discussed here, the workflow... huge simplification of the codebase. I still think this is the best choice, by far.

And as I have said before, if that's how you want to do things, if other maintainers that have an opinion on this agree with you than we can do it that way.

**Comment by jorgeamadosoria** [comment]
> And as I have said before, if that's how you want to do things, if other maintainers that have an opinion on this agree with you than we can do it that way.

Then I'll stop working on this one and implement PDF generation on the fly. Way easier and more economic than this orchestration effort. Feel free to close or disregard the corresponding pull: #2

**Comment by adamrusted** [comment]
Returning to this, following @jorgeamadosoria's on-the-fly attempt in the website repo. Would it make most sense for us to release this as an npm package, for the few that _do_ use the PDFs, therefore making the website generation much easier on our part. We can then use npm download stats to determine whether or not to keep this package going later on down the line.

**Comment by jorgeamadosoria** [comment]
> Returning to this, following @jorgeamadosoria's on-the-fly attempt in the website repo ([simple-icons/simple-icons-website#115](https://github.com/simple-icons/simple-icons-website/pull/115)). Would it make most sense for us to release this as an npm package, for the few that _do_ use the PDFs, therefore making the website generation much easier on our part. We can then use npm download stats to determine whether or not to keep this package going later on down the line.

If we provide the pdfs as a NPM package, what would that mean for the PDF repo?
I assume that a NPM package simple-icons-pdf won't affect the main package.
I'm ok with this idea in principle, but my main goal has not changed: I would like the pdf repo gone.

**Comment by ericcornelissen** [comment]
An npm package can be published from anywhere, so "what that means for the PDF repo" depends on how we setup the automation to push updates to the npm registry. If we decide to publish through this separate repository than it won't affect [the main repository](https://github.com/simple-icons/simple-icons) (except for the change of https://github.com/simple-icons/simple-icons/pull/6567) or [the main npm package](https://www.npmjs.com/package/simple-icons) for that matter.

If, as you want, we remove this repository than it would affect the main repository as it would have to take care of publishing to npm. That shouldn't affect the main npm package, but we would need to implement a suitable solution to publish multiple npm package from a single git repository.

**Pull request closed without merging**
